### PR TITLE
Trigger fetchMqlTimeSeries to run again if status is pending or running

### PR DESCRIPTION
**Pull request by courtneyholcomb**
# Changes
There is currently no logic triggering `useFetchMqlTimeSeries` to run again if the results are `PENDING` or `RUNNING`. The goal here is to move the useFetchMqlTimeSeries logic into `useEffect` with `queryReruns` as a dependency so that when `queryReruns` increments, useFetchMqlTimeSeries will be called again. 
Have not been able to test this change.

# Security Implications
* Please put any possible security-related concerns here. Feel free to @andykram if you're unsure. If none, please put "None"

### CODE REVIEW

File: hooks/useMqlQuery/useMqlQuery.ts [hunk 1]
```diff
@@ -44,27 +45,27 @@ export default function useMqlQuery({
...
-  useFetchMqlTimeSeries<CreateMqlQueryMutation>({
-    state,
-    skip: _skip,
-    dispatch,
-    createQueryIdQuery: createMqlQuery,
-    retries
-  });
+  }, [queryInput, metricName, mqlServerUrl, skip, queryReruns]);
 
+  if (state.queryStatus === MqlQueryStatus.Pending || state.queryStatus === MqlQueryStatus.Running) {
```
Comment (marcodamore) [inline hooks/useMqlQuery/useMqlQuery.ts:67]: Random thought... is it possible there's an in between state here that we're not picking up in these checks?

I.e. maybe another query status or the query status is undefined?

**Review by marcodamore — APPROVED** [review]

File: hooks/useMqlQuery/useMqlQuery.ts (hunk 1 above)
Comment (courtneyholcomb) [reply hooks/useMqlQuery/useMqlQuery.ts:67]: I don't think so, query status is directly pulled from the query result returned from the MQL server and we can see in the network responses that it's usually RUNNING when the retries stop.

File: hooks/useMqlQuery/useMqlQuery.ts [hunk 2]
```diff
@@ -44,27 +45,27 @@ export default function useMqlQuery({
...
+  const [queryReruns, setQueryReruns] = useState(0);
 
   useEffect(() => {
     if (!metricName || !mqlServerUrl || skip) {
       return;
+    } else if (!state.queryId) {
+      dispatch({ type: "postQueryStart" }); 
+      createMqlQuery({stateRetries: state.retries});
+    } else if (state.queryId && !state.cancelledQueries.includes(state.queryId)) {
+      useFetchMqlTimeSeries<CreateMqlQueryMutation>({
```
Comment (courtneyholcomb) [inline hooks/useMqlQuery/useMqlQuery.ts:57]: should add a setTimeout here or something so we poll more slowly

File: hooks/useMqlQuery/useMqlQuery.ts (hunk 1 above)
Comment (erikvorhes) [reply hooks/useMqlQuery/useMqlQuery.ts:67]: Yeah, there's no other state that we need to account for here. My theory is that there's no update to the parameters to the `useEffect`, so it fails to check again.

**Pull request closed without merging**